public/deck-selection.js

```diff
@@ -28,6 +28,39 @@ function setActiveTab(clickedButton) {
   }
 }
 
+/**
+ * Set up precon deck search functionality
+ * Filters deck tiles based on search input
+ */
+function setupPreconSearch() {
+  const searchInput = document.getElementById("precon-search");
+  if (!searchInput) return;
+
+  searchInput.addEventListener("input", function() {
+    const searchTerm = this.value.toLowerCase().trim();
+    const deckTiles = document.querySelectorAll(".precon-tile");
+
+    deckTiles.forEach(tile => {
```

Comment:
Don't compile the searchable text every time the input changes! Do that once during setup!